src/chart/gauge/GaugeView.ts

```diff
@@ -312,165 +324,296 @@ class GaugeView extends ChartView {
+                        group.remove(previousProgress);
+                        const progress = createProgress(newIdx, previousEndAngle) as graphic.Sector;
+                        const isClip = progressModel.get('clip');
+                        graphic.updateProps(progress, {
+                            shape: {
+                                endAngle: linearMap(data.get(valueDim, newIdx) as number, valueExtent, angleExtent, isClip)
+                            }
+                        }, seriesModel);
+                        group.add(progress);
+                        progressList[newIdx] = progress;
+                    }
+                })
+                .remove(function (idx) {
+                    if (showPointer) {
+                        const pointer = oldData.getItemGraphicEl(idx);
+                        group.remove(pointer);
+                    }
 
+                    if (showProgress) {
+                        const progress = oldProgressData[idx];
+                        group.remove(progress);
+                    }
+                })
+                .execute();
+
+            data.each(function (idx) {
+                const itemModel = data.getItemModel<GaugeDataItemOption>(idx);
+                const emphasisModel = itemModel.getModel('emphasis');
+                if (showPointer) {
+                    const pointer = data.getItemGraphicEl(idx) as PointerPath;
+                    pointer.useStyle(data.getItemVisual(idx, 'style'));
+                    pointer.setStyle(itemModel.getModel(['pointer', 'itemStyle']).getItemStyle());
+                    if (pointer.style.fill === 'auto') {
+                        pointer.setStyle('fill', getColor(
+                            linearMap(data.get(valueDim, idx) as number, valueExtent, [0, 1], true)
+                        ));
+                    }
+    
+                    (pointer as ECElement).z2EmphasisLift = 0
+                    setStatesStylesFromModel(pointer, itemModel);
+                    enableHoverEmphasis(pointer, emphasisModel.get('focus'), emphasisModel.get('blurScope'));
+                }
 
-            setStatesStylesFromModel(pointer, itemModel);
-            enableHoverEmphasis(pointer, emphasisModel.get('focus'), emphasisModel.get('blurScope'));
-        });
+                if (showProgress) {
+                    const progress = progressList[idx];
+                    progress.useStyle(data.getItemVisual(idx, 'style'));
+                    progress.setStyle(itemModel.getModel(['progress', 'itemStyle']).getItemStyle());
+                    setStatesStylesFromModel(progress, itemModel);
+                    enableHoverEmphasis(progress, emphasisModel.get('focus'), emphasisModel.get('blurScope'));
+                }
+            });
 
-        this._data = data;
+            this._data = data;
+            this._progressData = progressList;
+        }
+        else {
+            if (!showPointer) {
```

Comment:
I think there is no need to check `if (!showPointer)` here.  Since in this branch, `showPointer` and `showProgress` are all false.

Also, there is a `removeAll` code above. All the elements have been removed already. Also same issue in the above `add` `update`, `remove` callback